src/main/java/com/ericsson/ei/subscription/SubscriptionValidator.java

```diff
@@ -179,19 +190,19 @@ private static void RestPostMediaType(String restPostMediaType) throws Subscript
     }
 
     /**
-     * Validation of email address Throws SubscriptionValidationException if validation of the parameter
-     * fails due to wrong format of parameter.
+     * Validation of email address Throws SubscriptionValidationException if
+     * validation of the parameter fails due to wrong format of parameter.
      *
      * @param email
      */
     public static void validateEmail(String email) throws SubscriptionValidationException {
-        // When this regExp need to be changed then remember to change the one in the
-        // back-end (validEmailRegExpression in subscription.js), which do the same
-        // email validation check. The two
-        // regEx always need to be the same for ensuring the same check.
-        String validEmailRegExpression = "^(([^<>()\\[\\]\\\\.,;:\\s@\"]+(\\.[^<>()\\[\\]\\\\.,;:\\s@\"]+)*)|(\".+\"))@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}])|(([a-zA-Z\\-0-9]+\\.)+[a-zA-Z]{2,}))$";
-        final Pattern VALID_EMAIL_ADDRESS_REGEX = Pattern.compile(validEmailRegExpression,
-                Pattern.CASE_INSENSITIVE);
+        String validEmailRegEx = null;
+        try {
+            validEmailRegEx = (String) RegExProvider.getRegExs().get("validEmail");
```

Comment:
See above comment about casting.

Reply:
I thought the regex object in EiffelCommon was going to have same Key values as in Subscription object as in the Json object of regex. Which means an iterator can iterate the Subscription Json object and use that key to retrieve the corresponding regex from regex json object and do the regex. This make the code working no matter how many fields it is in Subscription json object or in Regex Object. If regex is added and/or if subscription fields added, no code changes need to be done, only Subscription Json file and Regex Json file need to updated with the new regex/subscription fields.

Reply:
But we are taking only two Subscription object values (Subscription name and email)to validate... So these values need to be called by name for retrieving regexs from common... or ...?

Reply:
I think he means calling the regex keys subscriptionName and notificationMeta instead of calling the m invalidName and validEmail. Though notificationMeta can be either a URL or email...

Reply:
I develop from a future perspective,,, so maintenance becomes as simple and fast as it get.
Your propose require more code to maintain and more code to change  when in future more regex is added or subscriptions fields added. Since you stated that meta field has several use case and different regex/checks, it is ok to have it like you have it. But change the field/regex key names according to what we decided.